demos/rag_agentic/notebooks/Level4_agentic_and_mcp.ipynb

```diff
@@ -0,0 +1,601 @@
+    "\n",
+    "### Installing dependencies\n",
+    "\n",
+    "This code requires `llama-stack` and the `llama-stack-client`, both at version `0.1.9`. Lets begin by installing them:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "4481ba68",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install llama-stack-client==0.1.9 llama-stack==0.1.9"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "62ee3cb2",
+   "metadata": {},
+   "source": [
+    "### Configuring logging\n",
+    "\n",
+    "Now that we have our dependencies, lets setup logging for the application:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "25fc0a44",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from llama_stack_client.lib.agents.event_logger import EventLogger\n",
+    "import logging\n",
+    "\n",
+    "logger = logging.getLogger(__name__)\n",
+    "if not logger.hasHandlers():  \n",
+    "    logger.setLevel(logging.INFO)\n",
+    "    stream_handler = logging.StreamHandler()\n",
+    "    stream_handler.setLevel(logging.INFO)\n",
+    "    formatter = logging.Formatter('%(message)s')\n",
+    "    stream_handler.setFormatter(formatter)\n",
+    "    logger.addHandler(stream_handler)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "681412e1",
+   "metadata": {},
+   "source": [
+    "### Connecting to llama-stack server\n",
+    "\n",
+    "For the llama-stack instance, you can either run it locally or connect to a remote llama-stack instance.\n",
+    "\n",
+    "#### Remote llama-stack\n",
+    "\n",
+    "- For remote, be sure to set `remote` to `True` and populate the `remote_llama_stack_endpoint` variable with your llama-stack remote.\n",
+    "- [Remote Setup Guide](https://github.com/opendatahub-io/llama-stack-on-ocp/tree/main/kubernetes)\n",
+    "\n",
+    "#### Local llama-stack\n",
+    "- For local, be sure to set `remote` to `False` and validate the `local_llama_stack_endpoint` variable. It is based off of the default llama-stack port which is `8321` but is configurable with your deployment of llama-stack.\n",
+    "- [Local Setup Guide](https://github.com/redhat-et/agent-frameworks/tree/main/prototype/frameworks/llamastack)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "6fa38ad0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "from dotenv import load_dotenv\n",
+    "load_dotenv()\n",
+    "\n",
+    "remote = os.getenv(\"REMOTE\").lower() in (\"true\", \"1\", \"yes\") # Use the `remote` variable to switching between a local development environment and a remote kubernetes cluster.\n",
```

Comment:
Is there a reason you are not just enforcing `REMOTE` to be a bool? I think it would be simpler to change this to  `"remote = os.getenv("REMOTE", True).`